README.md

```diff
@@ -57,8 +57,99 @@ you need to config the gem to run in sandbox mode:
+### Getting Started & Initiating Authentication Flow with Intuit
+
+What follows is an example using Rails but the principles can be adapted to any other framework / pure Ruby.
+
+Create a Rails initializer with:
+
+```ruby
+OAUTH_CONSUMER_KEY = ENV["OAUTH_CONSUMER_KEY"]
+OAUTH_CONSUMER_SECRET = ENV["OAUTH_CONSUMER_SECRET"]
+
+oauth_params = {
+  :site => "https://appcenter.intuit.com/connect/oauth2",
+  :authorize_url => "https://appcenter.intuit.com/connect/oauth2",
+  :token_url => "https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer"
+}
+
+::QB_OAUTH2_CONSUMER = OAuth2::Client.new(OAUTH_CONSUMER_KEY, OAUTH_CONSUMER_SECRET, oauth_params)
+
+```
+
+Your Controller action (the `grantUrl` above) should look like this:
+
+```ruby
+def authenticate
+  redirect_uri = quickbooks_oauth_callback_url
+  grant_url = ::QB_OAUTH2_CONSUMER.auth_code.authorize_url(:redirect_uri => redirect_uri, :response_type => "code", :state => SecureRandom.hex(12), :scope => "com.intuit.quickbooks.accounting")
+  redirect_to grant_url
+end
+```
+
+Where `quickbooks_oauth_callback_url` is the absolute URL of your application that Intuit should send the user when authentication succeeds. 
+
+That action should look like:
+
+```ruby
+def oauth_callback
+  if params[:state]
+    redirect_uri = oauth_callback_quickbooks_url
+    if resp = ::QB_OAUTH2_CONSUMER.auth_code.get_token(params[:code], :redirect_uri => redirect_uri)
+
+      # save your tokens here. For example:
+      # quickbooks_credentials.update_attributes(access_token: resp.token, refresh_token: resp.refresh_token, realm_id: params[:realmId])
+    end
+  end
+end
+```
+
+Most likely you will want to persist the OAuth access credentials so that users don't need to re-authorize your application in every session.
+
+An example database table would have fields likes:
+
+```sql
+access_token varchar(255),
+refresh_token varchar(255),
+realm_id varchar(255)
+```
+
+### Creating an OAuth Access Token
+
+Once you have your user's OAuth token, you can re-use the `OAuth Consumer` and create a `OAuth Client` using the `QB_OAUTH2_CONSUMER` you created earlier in your Rails initializer:
+
+```ruby
+qb_access_token = quickbooks_credentials.access_token
+qb_refresh_token = quickbooks_credentials.refresh_token
+
+access_token = OAuth2::AccessToken.new(::QB_OAUTH2_CONSUMER, qb_access_token, {refresh_token: qb_refresh_token})
+```
+
+### Access Token Validity and Token Refresh
+
+Each access token is only valid for one hour. The access token and refresh token can be refreshed directly by using OAuth Client:
+
+```ruby
+new_access_token = access_token.refresh!
+```
+
+The token must be assigned to a variable to prevent the loss of your new access token, which will void your credentials and a new set of credentials have to be acquired by authorizing the application again.
+Unauthorized (expired) access to the API will raise a `Quickbooks::AuthorizationFailure` error.
```

Comment:
If I try refreshing with a made up refresh token, the exception I get is `OAuth2::Error`:

```
OAuth2::AccessToken.new(QB_OAUTH2_CONSUMER, access_token, { refresh_token: 'abc' }).refresh!

OAuth2::Error: invalid_grant: Incorrect or invalid refresh token
{"error_description":"Incorrect or invalid refresh token","error":"invalid_grant"}
from /.../.bundle/gems/oauth2-1.4.1/lib/oauth2/client.rb:119:in `request'
```